core/src/sumeragi/main_loop.rs

```diff
@@ -138,48 +146,45 @@ impl Sumeragi {
     fn receive_network_packet(
         &self,
         view_change_proof_chain: &mut ProofChain,
-        control_message_in_a_row_counter: &mut usize,
-    ) -> Option<Message> {
+    ) -> (Option<BlockMessage>, bool) {
         const MAX_CONTROL_MSG_IN_A_ROW: usize = 25;
 
-        if *control_message_in_a_row_counter < MAX_CONTROL_MSG_IN_A_ROW {
-            *control_message_in_a_row_counter += 1;
-            self.control_message_receiver
+        let mut should_sleep = true;
+        for _ in 0..MAX_CONTROL_MSG_IN_A_ROW {
```

Comment:
why do we want to limit the number of sequential control flow messages?

Reply:
I think to prevent data message starvation in case of high frequency of service messages

Reply:
I get that, but is that necessary?

Reply:
yes imo e.g. malicious peer can spam with control flow messages